access.lua

```diff
@@ -262,6 +264,23 @@ and string.len(longest_skipped_match) > string.len(longest_unprotected_match) th
     return hlp.pass()
 end
 
+--
+-- 4b. Noauth URLs
+--
+-- If the URL matches one of the `no_auth` in the configuration file,
+-- it means that the URL should be protected by the SSO but no header
+-- has to be sent, even if the user is already authenticated.
+--
+
+if longest_noauth_match ~= ""
+and string.len(longest_noauth_match) >= string.len(longest_protected_match) 
+and string.len(longest_noauth_match) > string.len(longest_unprotected_match) then
+    if hlp.is_logged_in() then
+      logger.debug("Noauth "..ngx.var.uri)
+      return hlp.pass()
```

Comment:
I guess technically speaking this mechanism "works". But overall it seems like you're just duplicating the `skipped_uri` mechanics with a different name ? So my understanding is that that doesn't really address the original issue, which is about being unable to have the app working as expected if using `unprotected_uri` (or `protected_uri`, i.e. the app as 'private' with the new mechanism) because it will just be as if the app was skipped.

Instead, I believe this new `noauth_uri` thing should only be applied [here](https://github.com/YunoHost/SSOwat/blob/stretch-unstable/helpers.lua#L502) with something like : 

```
if [url is not in noauth_uris]
then
    -- Set the authorization header, except if the uri is in the noauth_uris
    ngx.req.set_header("Authorization", "Basic " [...etc...])
end
```

That way, the whole `noauth` thing remains independent from the whole protected/unprotected/(skipped) thing.

(Of course actually implementing it in lua is another thing as I'm not so much at ease with that language @.@)

Reply:
Yes the issue addressed is:
- when the app is private
- when the app is public and you are logged in YunoHost

Well its' a bit different from `skipped_uri`, with `skipped_uri` an app is always public, the content of the page is sent with `return hlp.pass()` without headers : https://github.com/YunoHost/SSOwat/blob/4f692c99d81164848975b32a501eaf688147c153/access.lua#L258-L263

In the case of noauth, content is only sent without headers or anything else when the user is logged in: L278 `if hlp.is_logged_in() then`

Maybe an array would help:
Usual way:
|         | Not Logged             | Logged              |
|---------|------------------------|---------------------|
| Public  | Content without header | Content with header |
| Private | Go to portal           | Content with header |

Using noauth: managing content without header when logged:
|         | Not Logged             | Logged                 |
|---------|------------------------|------------------------|
| Public  | Content without header | Content **without** header |
| Private | Go to portal           | Content **without** header |

&nbsp;

Your proposition was my first idea but as far I've seen and as I don't know lua, the only way would be to add arguments to the `function set_headers(user)`, like `function set_headers(user,uri,noauth_uris)` and it seems to me a bit to heavy as `set_headers(user)` is used several times in access.lua

Reply:
Alrighty, that makes more sense than I tought, sorry for not catching that subtlety first time.

Though imho we should still add an `and hlp.has_access()` to that `if` three lines above, as an app (or url) may be accessible only to specific users.

Apart from this, looks good to me :+1:

Reply:
That was very subtil and I didn't provided enough comments to make it understandable ... 
And that PR comes in the middle of permissions implementations.... No very easy ^^
I will look at the `and hlp.has_access()` and let you know

Reply:
`and hlp.has_access()` added